core/adapters/compare.go

```diff
@@ -0,0 +1,110 @@
+package adapters
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/smartcontractkit/chainlink/core/store"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/tidwall/gjson"
+)
+
+// Compare adapter type takes an Operator and a Value field to
+// compare to the previous task's Result.
+type Compare struct {
+	Operator string `json:"operator"`
+	Value    string `json:"value"`
+}
+
+var (
+	ErrResultNotNumber      = errors.New("The result was not a number")
+	ErrValueNotNumber       = errors.New("The value was not a number")
+	ErrOperatorNotSpecified = errors.New("Operator not specified")
+	ErrValueNotSpecified    = errors.New("Value not specified")
+)
+
+// Perform uses the Operator to check the run's result against the
+// specified Value.
+func (c *Compare) Perform(input models.RunResult, _ *store.Store) models.RunResult {
+	prevResult := input.Result()
+
+	if c.Value == "" {
+		input.SetError(ErrValueNotSpecified)
+		return input
+	}
+
+	switch c.Operator {
+	case "eq":
+		if c.Value == prevResult.String() {
+			input.CompleteWithResult(true)
+		} else {
+			input.CompleteWithResult(false)
+		}
```

Comment:
What about passing the result directly instead?  It would reduce the cognitive load for the statements.
i.e.
```
input.CompleteWithResult(c.Value == prevResult.String())
```